src/storage/sftp/storage.c

```diff
@@ -0,0 +1,931 @@
+        FUNCTION_TEST_PARAM(INT, rc);
+    FUNCTION_TEST_END();
+
+    ASSERT(this != NULL);
+
+    FUNCTION_TEST_RETURN(
+        BOOL, rc == LIBSSH2_ERROR_SFTP_PROTOCOL && libssh2_sftp_last_error(this->sftpSession) == LIBSSH2_FX_NO_SUCH_FILE);
+}
+
+/**********************************************************************************************************************************/
+static StorageInfo
+storageSftpInfo(THIS_VOID, const String *const file, const StorageInfoLevel level, const StorageInterfaceInfoParam param)
+{
+    THIS(StorageSftp);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(STORAGE_SFTP, this);
+        FUNCTION_LOG_PARAM(STRING, file);
+        FUNCTION_LOG_PARAM(ENUM, level);
+        FUNCTION_LOG_PARAM(BOOL, param.followLink);
+    FUNCTION_LOG_END();
+
+    FUNCTION_AUDIT_STRUCT();
+
+    ASSERT(this != NULL);
+    ASSERT(file != NULL);
+
+    StorageInfo result = {.level = level};
+
+    // Stat the file to check if it exists
+    LIBSSH2_SFTP_ATTRIBUTES attr;
+    int rc;
+    Wait *const wait = waitNew(this->timeoutConnect);
+
+    do
+    {
+        rc = libssh2_sftp_stat_ex(
+            this->sftpSession, strZ(file), (unsigned int)strSize(file), param.followLink ? LIBSSH2_SFTP_STAT : LIBSSH2_SFTP_LSTAT,
+            &attr);
+    }
+    while (rc == LIBSSH2_ERROR_EAGAIN && waitMore(wait));
+
+    waitFree(wait);
+
+    // Throw libssh2 errors other than no such file
+    if (rc != 0)
+    {
+        if (!storageSftpLibSsh2FxNoSuchFile(this, rc))
+            THROW_FMT(FileOpenError, STORAGE_ERROR_INFO, strZ(file));
+    }
+    // Else the file exists
+    else
+    {
+        result.exists = true;
+
+        // Add type info (no need set file type since it is the default)
+        if (result.level >= storageInfoLevelType && !LIBSSH2_SFTP_S_ISREG(attr.permissions))
+        {
+            if (LIBSSH2_SFTP_S_ISDIR(attr.permissions))
+                result.type = storageTypePath;
+            else if (LIBSSH2_SFTP_S_ISLNK(attr.permissions))
+                result.type = storageTypeLink;
+            else
+                result.type = storageTypeSpecial;
+        }
+
+        // Add basic level info
+        if (result.level >= storageInfoLevelBasic)
+        {
+            if ((attr.flags & LIBSSH2_SFTP_ATTR_ACMODTIME) != 0)
+                result.timeModified = (time_t)attr.mtime;
+
+            if (result.type == storageTypeFile)
+                if ((attr.flags & LIBSSH2_SFTP_ATTR_SIZE) != 0)
+                    result.size = (uint64_t)attr.filesize;
+        }
+
+        // Add detail level info
+        if (result.level >= storageInfoLevelDetail)
+        {
+            if ((attr.flags & LIBSSH2_SFTP_ATTR_UIDGID) != 0)
+            {
+                result.groupId = (unsigned int)attr.gid;
+                result.group = groupNameFromId(result.groupId);
+                result.userId = (unsigned int)attr.uid;
+                result.user = userNameFromId(result.userId);
+            }
```

Comment:
I don't think groupNameFromId() and userNameFromId() are going to work as expected. These will map user/groups on the host running the backup, not the SFTP host.

Reply:
I've removed the setting of group and user.  commit e00b39b8542bbe0b6fdc3ff0e569fd745d0f55b7